plan/optimizer.go

```diff
@@ -82,7 +82,6 @@ func doOptimize(logic LogicalPlan, ctx context.Context, allocator *idAllocator)
 	}
 	pp := info.p
 	pp = EliminateProjection(pp)
-	physicalInitialize(pp)
 	addCachePlan(pp, allocator)
```

Comment:
1. `Cache` Plan is a PhysicalPlan, we can only do it after physicalplanbuilder
2. The order of `EliminateProj` and `addCachePlan` is arbitrary.